src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -1024,7 +1024,23 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
                         attributeValue = useViewData ? EvalString(viewContext, fullName, format) : valueParameter;
                     }
 
-                    tagBuilder.MergeAttribute("value", attributeValue, replaceExisting: isExplicitValue);
+                    var addValue = true;
+                    object typeAttributeValue = null;
```

Comment:
hmm, used only if `TryGetValue` succeeds. if the compiler lets you, remove the initialization.